File: trunk/C#/RBD/RBD/TPC/Cor/WaitingState.cs
// +

using System.Net;
using RBD.TPC.Msg;

using RBD.Msg.Client;

namespace RBD.TPC.COR
{
    public class WaitingState : CoordinatorState
    {
        private static string LOGGING_NAME = "Coordinator.WaitingState";

        override public void onTimeout()
        {
            coordinator.abortTransaction(new RBD.Msg.Client.TimeoutMessage());
        }

        override public void onYesForCommit(IPEndPoint node)
        {
            coordinator.processAnswer(node, new PreCommitMessage(), new PreparedState());
            Logger.getInstance().log("Got answer "
 				+ coordinator.getAnswerCount() + " of "
 				+ coordinator.getNodesCount(), LOGGING_NAME,
 				Logger.Level.INFO);
        }

        override public void onNoForCommit(IPEndPoint node)
        {
            Logger.getInstance().log("Got no - aborting!", LOGGING_NAME, Logger.Level.INFO);
            coordinator.abortTransaction(new ConflictMessage());
        }
    }
}

File: java/SR-RBD/src/ddb/tpc/cor/WaitingState.java
/**
 * 
 */
package ddb.tpc.cor;

import java.net.InetSocketAddress;

import ddb.Logger;
import ddb.msg.client.ConflictMessage;
import ddb.msg.client.TimeoutMessage;
import ddb.tpc.TimeoutListener;
import ddb.tpc.msg.PreCommitMessage;

/**
 */
public class WaitingState extends CoordinatorState {
	
	private final static String LOGGING_NAME = "Coordinator.WaitingState";
	
	/** 
	 * 
	 * @see TimeoutListener#onTimeout()
	 * 
	 */
	@Override
	public void onTimeout() {
		coordinator.abortTransaction(new TimeoutMessage());
	}

	/** 
	 *  @see CoordinatorState#onYesForCommit(String node)
	 */
	@Override
	public void onYesForCommit(InetSocketAddress node) {
	
		coordinator.processAnswer(node, new PreCommitMessage(), new PreparedState());
		Logger.getInstance().log("Got answer " 
			+ coordinator.getAnswerCount() + " of " 
			+ coordinator.getNodesCount(), LOGGING_NAME, 
			Logger.Level.INFO);
	}

	/** 
	 *  @see CoordinatorState#onNoForCommit(String node)
	 * 
	 */
	@Override
	public void onNoForCommit(InetSocketAddress node) {
		Logger.getInstance().log("Got no - aborting!", LOGGING_NAME, Logger.Level.INFO);
		coordinator.abortTransaction(new ConflictMessage());
	}
}